builtin/credential/approle/path_role.go

```diff
@@ -857,55 +857,79 @@ func (b *backend) pathRoleRead(req *logical.Request, data *framework.FieldData)
 
 	lock := b.roleLock(roleName)
 	lock.RLock()
-	defer lock.RUnlock()
+	writeLockHeld := false
```

Comment:
Rather than this bool, since the signatures are the same, you can just create a function like `lockRelease := lock.RUnlock` and below if you switch to a write lock you can reassign to `lockRelease = lock.Unlock`. THen below you just call `lockRelease()`.

Reply:
Good suggestion. Done!